### PR TITLE
JENA-1881: Model API changes for RDF*

### PR DESCRIPTION
**Pull request by afs**
This PR adds triple term to the mode API by making them a kind of "Resource", along side URI and blank node.

This is the only way to incorporate triple terms without a major change to the type-safe API because they can appear in the subject position.

### CODE REVIEW

File: jena-core/src/main/java/org/apache/jena/rdf/model/ResourceFactory.java [hunk 1]
```diff
@@ -92,6 +92,15 @@ public static Resource createResource(String uriref) {
         return instance.createResource(uriref);
     }
     
+    /** 
+     * create a new resource representing an RDF* triple term.
```
Comment (kinow) [inline jena-core/src/main/java/org/apache/jena/rdf/model/ResourceFactory.java:96]: s/create/Create

File: jena-core/src/main/java/org/apache/jena/rdf/model/ResourceFactory.java [hunk 2]
```diff
@@ -212,6 +221,13 @@ public static Statement createStatement(
          */
         public Resource createResource(String uriref);
         
+        /** create a new resource representing an RDF* triple term.
```
Comment (kinow) [inline jena-core/src/main/java/org/apache/jena/rdf/model/ResourceFactory.java:224]: Uppercase create? (Other methods seem to start with upper case...)

File: jena-core/src/main/java/org/apache/jena/rdf/model/ResourceFactory.java (hunk 1 above)
Comment (afs) [reply jena-core/src/main/java/org/apache/jena/rdf/model/ResourceFactory.java]: Many of the javadoc comments are like that.  ("Answer" but "create")

Redid the commit : I've gone through the javadoc on all methods and tidied up.

Comment (kinow) [reply jena-core/src/main/java/org/apache/jena/rdf/model/ResourceFactory.java]: Sorry, I had looked only at a couple comments available in the GitHub interface.

Comment (afs) [reply jena-core/src/main/java/org/apache/jena/rdf/model/ResourceFactory.java]: No problem - usual balance between focused changes for the ticket addressed and a desire to sort out things in passing.

**Pull request merged**